[CP-24464] default to create self-signed cert upon chart install (#130)
Cloudzero/cloudzero-charts: charts/cloudzero-agent/README.md

For an optimal installation experience, we recommend the following:

- Have a basic understanding of Kubernetes and Helm charts.
- Be prepared with a list of the labels and annotations your organization plans to collect if you don't want the default: all pod and namespace labels with the key `app.kubernetes.io/component`. See [Labels and Annotations](#labels-and-annotations).

## Installation

### Quick Start

> > ⚠️ By default, this chart exports all pod and namespace labels with keys matching `app.kubernetes.io/component`. This behavior can be changed as needed; see the [Labels and Annotations](#labels-and-annotations) section for details.

Reviewer: ```suggestion
> ⚠️ By default, this chart exports only the `app.kubernetes.io/component` _label_ for pods and namespace. No annotations are exported by default. This is likely insufficient for your Organization, but provides a safe default for demo purposes. `labels` are important to configure to according to your Organization's FinOps tagging practices. It is recommended to read the [Labels and Annotations](#labels-and-annotations) section for details on exposing an expanded list of label, or annotations for your Organization's need.
```

#### 1. Add CloudZero Helm Repository
Refer to the [`helm repo`](https://helm.sh/docs/helm/helm_repo/) documentation for command details. To use a beta version, refer to the [beta installation document](./BETA-INSTALLATION.md) for the appropriate channel.

```console
helm repo add cloudzero https://cloudzero.github.io/cloudzero-charts
helm repo update
```

#### 2. Install Helm Chart

```console
helm install <RELEASE_NAME> cloudzero/cloudzero-agent \
    --set apiKey=<CLOUDZERO_API_KEY> \
    --set clusterName=<CLUSTER_NAME> \
    --set-string cloudAccountId=<CLOUD_ACCOUNT_ID> \
    --set region=<REGION>
```

> ### Advanced Install

Reviewer: This adds a visual break in the document

- Have a basic understanding of Kubernetes and Helm charts.
- Be prepared with a list of the labels and annotations your organization plans to collect.

## Installation

### Easy Install

To use the chart or a beta version, you must add the repository to Helm. Refer to the [`helm repo`](https://helm.sh/docs/helm/helm_repo/) documentation for command details.

```console
helm repo add cloudzero https://cloudzero.github.io/cloudzero-charts
helm repo update
```

#### 1. Install Helm Chart

```console
helm install <RELEASE_NAME> cloudzero/cloudzero-agent \
    --set existingSecretName=<NAME_OF_SECRET> \
    --set clusterName=<CLUSTER_NAME> \
    --set-string cloudAccountId=<CLOUD_ACCOUNT_ID> \

>     --set region=<REGION>

Author: I am not sure where we landed on requiring users to know what labels to allow. we have most recently failed the installation early if the user does not make a choice about whether they want to collect labels, and which labels to collect. but, there was some discussion about removing that. 

the only downsides of requiring the installer to know those things is:
1. it makes the installation a little uglier. the user would probably have to create a yaml file like:
```yaml
insightsController:
  labels:
    enabled: true
    patterns:
      - '.*'
```
which kind of defeats the purpose of the "easy install"
2. a lot of times, the installers are not the ultimate users, so they just don't know what to put

on the other hand, if we make a "sane default", the risk is that the chart will get installed, and then a week later, other users will complain that we're not exporting the labels they want

and finally, another option is to allow all labels by default. but, I think that's not such a good idea because we could get a ton of labels and overwhelm the tag limit

Author: @teevans @josephbarnett , thoughts?

Reviewer: What if we have that by default, and in a callout (`> ⚠️` section) which states there are limits on labels and they should adjust this value in production environments to restrict the value to those needed by their Teams.

ie. for first run/testing install, I think that is a sane default, but for the DevOps team - they may need to see the warning symbol to trigger questions on what is the right values...

Reviewer: We might also name it `Quick Start` version easy install. But that is my OCD as a dev...

- Have a basic understanding of Kubernetes and Helm charts.

> - Be prepared with a list of the labels and annotations your organization plans to collect.

Reviewer 2: ```suggestion
- Be prepared with a list of the labels and annotations your organization plans to collect if you don't want the default: all Pod and Namespace labels with the key `app.kubernetes.io/component`. See [Labels and Annotations](#labels-and-annotations).
```

## Installation

> ### Quick Start

Reviewer 2: ```suggestion
### Quick Start

> ⚠️ By default, this chart exports **all** Pod and Namespace labels with keys matching `app.kubernetes.io/component`. This can result in a large quantity of labels, and only the first 300 will be shown in CloudZero. [Ensure defaults are adjusted if necessary](#labels-and-annotations) **before** installation, or you will need to update and redeploy the chart to make changes.
```

Author: I like this placement

I think the "danger" is more that they may not get the labels they expect, and less so that the limit will be hit. mind if I reword this a bit to reflect that?

Reviewer 2: Good catch! That would be great, thank you!

#### 1. Add CloudZero Helm Repository
Refer to the [`helm repo`](https://helm.sh/docs/helm/helm_repo/) documentation for command details. To use a beta version, refer to the [beta installation document](./BETA-INSTALLATION.md) for the appropriate channel.

```console
helm repo add cloudzero https://cloudzero.github.io/cloudzero-charts
helm repo update
```

#### 2. Install Helm Chart

```console
helm install <RELEASE_NAME> cloudzero/cloudzero-agent \
    --set apiKey=<CLOUDZERO_API_KEY> \
    --set clusterName=<CLUSTER_NAME> \
    --set-string cloudAccountId=<CLOUD_ACCOUNT_ID> \
    --set region=<REGION>
```

### Advanced Install

> While the "Easy Install" option will cover most cases, the chart has many options for customization.

Reviewer: I might add `most demo use cases - but may not be appropriate for a production deployment`.

This section provides configuration options to ensure production quality deployment...

| apiKey            | string | `nil`                 | The CloudZero API key to use for exporting metrics. Only used if `existingSecretName` is not set.                       |
| existingSecretName| string | `nil`                 | Name of the secret that contains the CloudZero API key. Required if not providing the API key via `apiKey`.             |
| region            | string | `nil`                 | Region where the cluster is running (e.g., `us-east-1`, `eastus`). For more information, see AWS or Azure documentation. |

> 

Reviewer: I can't help but wonder if this should be listing only what is absolutely mandatory, such as the key, account, clustername, region, and point to a configuration values readme with the fill list of variables?

I don't think this has to be done now, just thinking out loud.

#### Overriding Default Values

> 

Reviewer: I wonder if you could break this whole section out to docs/helm-cheatsheet.md file, and point to it in the "## Useful References" section?

### Labels and Annotations

> ⚠️ CloudZero supports a maximum of **300 labels** for Kubernetes resources. Ensure you configure regex patterns to gather only the necessary labels/annotations. Additional labels after the first 300 are discarded.

> **By default**, this chart exports pod and namespace labels with keys matching `app.kubernetes.io/component`, and no annotations. You can configure what labels and/or annotations exported by following the steps in this section.

Reviewer 2: ```suggestion
**By default**, this chart exports pod and namespace labels with keys matching `app.kubernetes.io/component`, and no annotations. You can configure what labels and/or annotations are exported by following the steps in this section.
```

> - `Node`

Reviewer 2: I'd suggest adding this to the beginning of the `Labels and Annotations` section a few lines up:

```
> ⚠️ CloudZero supports a maximum of **300 labels** for Kubernetes resources. Ensure you configure regex patterns to gather only the necessary labels/annotations. Additional labels after the first 300 are discarded.

**By default**, this chart exports **all** Pod and Namespace labels with keys matching `app.kubernetes.io/component`, and no annotations. You can configure what labels and/or annotations exported by following the steps in this section.
```

Then the next part would be "This chart allows the exporting..."

Author: added 👍

- `Namespace`

The export of labels and annotations from a cluster can be turned on or off within the `insightsController` field. For example, the following enables exporting both labels and annotations from pods and namespaces:
```yaml
insightsController:
  enabled: true
  labels:
    enabled: true
  annotations:
    enabled: true
```

It is recommended to supply a list of regexes to filter only the labels/annotations required:
```yaml
insightsController:
  enabled: true
  labels:
    enabled: true
    patterns:
      - '^foo' # -- Match all labels whose key starts with "foo"
      - 'bar$' # -- Match all labels whose key ends with "bar"
```

> Labels/annotations can also be gathered by more than just pods and namespaces. An example of gather labels from all available resources would be:

Reviewer 2: ```suggestion
Labels/annotations can also be gathered by more than just pods and namespaces. An example of gathering labels from all available resources would be:
```

Author: updated ✅
